packages/flame/lib/src/components/position_component.dart

```diff
@@ -135,20 +206,106 @@ abstract class PositionComponent extends BaseComponent {
+      final m = _transformMatrix.storage;
+      final cosA = math.cos(_angle);
+      final sinA = math.sin(_angle);
+      final deltaX = -anchor.x * _size.x;
+      final deltaY = -anchor.y * _size.y;
+      m[0] = cosA * _scaleX;
+      m[1] = sinA * _scaleX;
+      m[4] = -sinA * _scaleY;
+      m[5] = cosA * _scaleY;
+      m[12] = _position.x + m[0] * deltaX + m[4] * deltaY;
+      m[13] = _position.y + m[1] * deltaX + m[5] * deltaY;
+      _recalculateTransform = false;
+    }
+    return _transformMatrix;
+  }
+
+  /// Transform `point` from local coordinates into the parent coordinate space.
+  Vector2 localToParent(Vector2 point) {
+    final m = transformMatrix.storage;
+    return Vector2(
+      m[0] * point.x + m[4] * point.y + m[12],
+      m[1] * point.x + m[5] * point.y + m[13],
+    );
+  }
+
+  /// Transform `point` from local coordinates into the global (screen)
+  /// coordinate space. For example, local point (0, 0) corresponds to
+  /// the top left corner of the component. Thus,
+  /// `localToAbsolute(Vector2(0, 0))` returns the screen coordinates
+  /// of the top left corner of the component.
+  Vector2 localToAbsolute(Vector2 point) {
+    var c = parent;
+    while (c != null) {
+      if (c is PositionComponent) {
+        return c.localToAbsolute(localToParent(point));
+      }
+      c = c.parent;
+    }
+    return localToParent(point);
+  }
+
+  /// Transform `point` from the parent's coordinate space into the local
+  /// coordinates.
+  Vector2 parentToLocal(Vector2 point) {
+    // Here we rely on the fact that in the transform matrix only elements
+    // `m[0]`, `m[1]`, `m[4]`, `m[5]`, `m[12]`, and `m[13]` are modified.
+    // This greatly simplifies computation of the inverse matrix.
+    final m = transformMatrix.storage;
+    final det = m[0] * m[5] - m[1] * m[4];
+    return Vector2(
+      ((point.x - m[12]) * m[5] - (point.y - m[13]) * m[4]) / det,
+      ((point.y - m[13]) * m[0] - (point.x - m[12]) * m[1]) / det,
+    );
+  }
+
+  /// Transform `point` from the global (screen) coordinate space into the
+  /// local coordinates. This can be used, for example, to detect whether
+  /// a specific point on the screen lies within this `PositionComponent`,
+  /// and where exactly it is.
+  Vector2 absoluteToLocal(Vector2 point) {
+    var c = parent;
+    while (c != null) {
+      if (c is PositionComponent) {
+        return parentToLocal(c.absoluteToLocal(point));
+      }
+      c = c.parent;
+    }
+    return parentToLocal(point);
+  }
 
+  /// Test whether the `point` (given in global coordinates) lies within this
+  /// component. The top and the left borders of the component are inclusive,
+  /// while the bottom and the right borders are exclusive.
   @override
   bool containsPoint(Vector2 point) {
-    final rectangle = Rectangle.fromRect(toAbsoluteRect(), angle: angle)
-      ..position = absoluteCenter;
-    return rectangle.containsPoint(point);
+    final local = absoluteToLocal(point);
```

Comment:
This does not take angle into consideration

Reply:
Perhaps it is not that obvious, but `absoluteToLocal()` chains `parentToLocal()` along the path to the root of the component tree. At the same time, `parentToLocal()` applies an inverse of the `_transformMatrix` to the given `point`. Thus, this method takes into account all the transforms that were composed within the transform matrix: translations, rotations, scaling, and flips.

Reply:
Aah makes sense! Could you add that to the dartdoc comments? (if it is not already there)
Someone asked for these types of methods on discord the other day.